### ISSUE TITLE
Can this be extended to support expression tags in apache config?

### ISSUE THREAD

**Issue by agrawalo** [etingof/apacheconfig#13]
I am not sure whether this parser can be extended to support expression tags, please comment. Something like jsp expression tags or something in these lines http://jinja.pocoo.org/

**Comment by etingof** [comment]
The expression tags that seem conceptually close to Apache config would probably be [ap_expr](https://httpd.apache.org/docs/2.4/expr.html). But these seem quite rigid in what they can do and httpd-specific.

[jsp](https://www.javatpoint.com/jsp-expression-tag) and jinja expression tags seem more functional e.g. you can even embed and eval some code there.

WDYT?

**Comment by etingof** [comment]
It seems that it should be easy to parse stuff out of `<>` tag and just feed it to the jinja engine which would take care of the rest.

**Comment by agrawalo** [comment]
Thanks :) then this library will become super powerful I guess.

**Comment by agrawalo** [comment]
Do you have such a use-case in your application for which you created this parser?

**Comment by etingof** [comment]
No, my config files do not happen to use expression tags of any kind. But I think I could try to add those for the greater good. Or you can come up with a PR. ;-)

**Comment by etingof** [comment]
One thing I do not quite get is your use-case... Do you need this parser to actually evaluate expression tags? Or just parse/dump them untouched?

I wonder how reasonable would be to evaluate the expression tags given they tend to rely upon HTTP request runtime variables....

**Comment by agrawalo** [comment]
>> Do you need this parser to actually evaluate expression tags? 
I guess this would be bit difficult but this can be configurable too (for some straight forward configs users can opt for an option to evaluate expressions inside the parser itself). 

>> how reasonable would be to evaluate the expression tags given they tend to rely upon HTTP request runtime variables
It doesn't have to an HTTP request runtime variables everytime.

Evaluating expressions gives users flexibility to generate dynamic configurations with less config lines. For example users can write loops or write conditional statements.

**Comment by etingof** [comment]
> Evaluating expressions gives users flexibility to generate dynamic configurations with less config lines. For example users can write loops or write conditional statements. Let me know if you think otherwise.

Hold on, are you talking about using this parser with the new projects, not related to existing Apache-like configs? Because original Apache config file syntax does not seem to support loops, just conditionals. Is that correct?

**Comment by agrawalo** [comment]
correct.

**Comment by etingof** [comment]
BTW, the parser in its current state can parse `ap_expr` expression tags. Just parse and dump, not evaluate. So I added a test case in commit 779fef8dc703a15644e6bf5ac9eeb02b564218db

**Comment by agrawalo** [comment]
Not sure if you would want to check https://docs.saltstack.com/en/latest/ref/renderers/
